example/linux/installation.md

```diff
@@ -0,0 +1,66 @@
+# Talawa API Installation Guide
+
+This guide provides step-by-step instructions for setting up the Talawa API service on a Linux system using systemd.
+
+## Prerequisites
+
+- Node.js (version 20.18.0 or current version the app is running on) (Ensure the Linux Node.js version matches the app version)
+- nvm (Node Version Manager)
+- tsx (TypeScript execution environment should be installed globally. Command : npm install -g tsx )
+- A Linux system with systemd
+- For development, first run manually to see if it is working. Check the `.env` file to ensure `NODE_ENV` is set to `development`.
+- For production, first build the app to generate the `dist` folder, then set `NODE_ENV` to `production`.
+- Ensure you provide the correct paths for each configuration to avoid errors.
+- The service dynamically starts the development and production server based on the `.env` configuration, so verify it before making any changes.
+- Always check the log files to identify errors.
+
+## Steps
+
+1. **Create the Systemd Service File**:
+   - Create a file named `talawa-api.service` in the `/etc/systemd/system` directory. Ensure you have root privileges.
+
+2. **Edit the `talawa-api.service` File**:
+   - Modify the `talawa-api.service` file as required (e.g., `ExecStart`, `WorkingDirectory`, `User`, `Group`). You can find how to edit these fields in the `talawa-api.service` file located in `example/linux/systemd`.
+
+3. **Copy the Configuration**:
+   - Copy the text into the `talawa-api.service` file located in `/etc/systemd/system` and save it.
+
+4. **Edit the `Talawa-api.sh` Script**:
+   - Edit the `Talawa-api.sh` script to set the project directory and log file path:
+   
+     ```bash
+     # Change to the project directory
+     cd /usr/local/talawa-api
+
+     # Define log file
+     LOG_FILE="/var/log/talawa-api.log"
+     ```
+
```

Comment:
[Unit]
# Description of the service
Description=Talawa-API Service

# Ensure the service starts after the network is available
After=network.target

[Service]
# The type of service. 'simple' means the service will start immediately.
Type=simple

# The command to start the service. This points to the Talawa-api.sh script.User have to set it manually. Inside the talawa-api directory execute the command "pwd" to find the path.Then replace this with your original one (eg:/usr/local/talawa-api/example/linux/systemd/Talawa-api.sh -> /home/purnendu/development/talawa-api/example/linux/systemd/Talawa-api.sh)
ExecStart=/usr/local/talawa-api/example/linux/systemd/Talawa-api.sh

# The working directory for the service.User have to set it manually.Inside the talawa-api directory execute the command "pwd" to find the path.Then replace this with your original one (eg:/usr/local/talawa-api -> /home/purnendu/development/talawa-api)
WorkingDirectory=/usr/local/talawa-api

# Restart the service automatically if it stops.
Restart=always

# The delay before restarting the service.
RestartSec=5

# The user to run the service as. You can find your username by running 'whoami'.Create a user named talawa for better understanding and Security
User=talawa

# The group to run the service as. Usually, this is the same as the username.
Group=talawa

# Redirects the output and error to the systemd journal and console.
StandardOutput=journal+console
StandardError=journal+console

# Sets the maximum number of open files.Neither too much nor too less.
LimitNOFILE=15000

# Security-related directives

# Protect the system by making the root filesystem read-only for the service.
# This prevents the service from modifying system-critical files, reducing the attack surface.
ProtectSystem=full

# Prevent the service from accessing home directories of other users.
# Only the necessary files in the service's user directory will be accessible.
ProtectHome=yes

# Make the service's working directory read-only.
# This is useful for protecting important directories from being altered by the service.
ReadOnlyPaths=/usr/local/talawa-api

# Ensures the service and its child processes cannot gain new privileges.
# This helps to prevent privilege escalation attacks.
NoNewPrivileges=true

# Isolate the service's temporary files from others by assigning it a private /tmp.
# This prevents other services or users from accessing or interfering with the service's temporary data.
PrivateTmp=true

# Restrict the service to use only IPv4 and IPv6 address families.
# This helps prevent the service from using other network protocols, such as Unix domain sockets, which could be a potential attack vector.
RestrictAddressFamilies=AF_INET AF_INET6

# Allows the service to bind to network ports below 1024 (e.g., HTTP on port 80), 
# but it won't allow the service to use any other privileged capabilities.
AmbientCapabilities=CAP_NET_BIND_SERVICE

[Install]
# Specifies the target to which the service should be added. 'multi-user.target' means the service will start in multi-user mode.
WantedBy=multi-user.target
now verify this section now i have updated with your suggestions to improve security and file descripter also managed and give comment for user to add the path in it manually should be fine because it is a one time process once it starts then autometically it will run.

Reply:
ok actually what are the suggestion u gave it will be mentioned in installation.md file